compose/project.py

```diff
@@ -284,12 +284,31 @@ def build(self, service_names=None, no_cache=False, pull=False):
             else:
                 log.info('%s uses an image, skipping' % service.name)
 
+    def clean(self, keep=False):
+        all_containers = (self.containers(orphan=True, stopped=True) +
+                          self.containers(orphan=True, one_off=True))
+        running_containers = [c for c in all_containers if c.is_running]
+        parallel_execute(
+            objects=running_containers,
+            obj_callable=lambda c: c.kill(),
+            msg_index=lambda c: c.name,
+            msg="Killing"
+        )
+        if not keep:
+            parallel_execute(
+                objects=all_containers,
+                obj_callable=lambda c: c.remove(),
+                msg_index=lambda c: c.name,
+                msg="Removing"
+            )
```

Comment:
It would be good to use the same `parallel_execute()` for both tasks, but not entirely necessary.